### ISSUE TITLE
Add a property to XAML TextBlocks for enabling a tooltip for TextBlock text only when IsTextTrimmed is true

### ISSUE THREAD

**Issue by shufan** [microsoft/microsoft-ui-xaml#170]
# Proposal: Add a property to XAML TextBlocks for enabling a tooltip for TextBlock text only when IsTextTrimmed is true

## Summary
This property when set to True on a TextBlock would set a tooltip for the TextBlock with the text of the TextBlock as the tooltip content, when and only when the TextBlock's IsTextTrimmed property is True.

## Rationale
With the addition of the Text Scaling Ease of Access setting, oftentimes TextBlocks will get clipped/trimmed with Text Scaling set to something like 225%. The [recommended guidance](https://docs.microsoft.com/en-us/windows/uwp/design/input/text-scaling) for dealing with this when space for the TextBlock is limited is to add a ToolTip. However, using ToolTipService.ToolTip="Tooltip text" results in a tooltip that is always visible, even when the TextBlock is not trimmed. As a result, helpers have been scattered around the Windows codebase to help implement this sort of "auto tooltip" as an attached behavior. It seems like this should really be made a built-in feature of TextBlocks. This will be especially helpful for Text Scaling accessibility but helpful even in other cases where real estate can be limited dynamically.

## Usage Examples
_excuse the placeholder property name_
>\<TextBlock
>&nbsp;&nbsp;&nbsp;&nbsp;Text="{Binding DisplayName}"
>&nbsp;&nbsp;&nbsp;&nbsp;TextTrimming="CharacterEllipsis"
>&nbsp;&nbsp;&nbsp;&nbsp;ToolTipOnTextTrimmed="True"/>

## Open Questions

1. Should this be implemented as an attached behavior provided by ToolTipService?
2. If implemented as a property on TextBlock should the property default to False or True?
3. How does this affect control derived from TextBlock or tooltips set on parent elements?

**Comment by bschoepke** [comment]
The Start menu team built a similar component; definitely should be in the framework.

**Comment by micahl** [comment]
Any reason that it should not be True by default given that its the recommended guidance?  So, rather than be opt-in have it be an opt-out (assuming a default opt-in didn't adversely impact performance - TBD).

**Comment by mdtauk** [comment]
Perhaps if it is to be added, it be on by default, unless the developer has attached their own tooltip to the control to override the default behaviour.  Maybe an empty value for disabling all tooltips.

**Comment by shufan** [comment]
Yeah thats a good point, seems reasonable to have it on by default. I'm assuming a tooltip set on the TextBlock or a parent element would override the auto tooltip for trimmed text?

**Comment by mdtauk** [comment]
> Yeah thats a good point, seems reasonable to have it on by default. I'm assuming a tooltip set on the TextBlock or a parent element would override the auto tooltip for trimmed text?

I do not think the UWP platform has the concept of differentiating tooltips based on the control's current state.  A developer could do it, but how to introduce it as a concept in the control, and whether that would extend to controls derived from it like the SearchBox and EditableComboBoxes?

Would it require a behaviour property to turn it on or off.  Would it need to have two tooltips attached, one for when trimmed, one for other circumstances?

**Comment by lukasf** [comment]
Oh yeah. I also have one of those helpers. This would be very useful. And it might be possible to do this in a more performant way, when directly adding it into TextBlock. Then, duplicate measure on the rendered text size could perhaps be avoided.

Currently, I only enable my behavior explicitly on selected, important fields. For many fields it is not needed (they cannot shrink due to layout), plus my solution involves an extra measure pass. If the implementation can be done internally, without performance penalty, then I would agree it could be enabled by default. But if it has a performance penalty, it sure is better to keep it off by default. Penalty can easily multiply when using in list layouts.

I do not think that we need two ToolTips attached. The manual tooltip should always have preference. Only use the auto tooltip if none is assigned.

**Comment by marb2000** [comment]
@shufan  Thank you so much for this feature request. Unfortunately, given the huge amount of work this feature request won't be addressed soon. However, we do see the value and we will keep it in the backlog for future consideration. @bschoepke, @mdtauk, @lukasf, and @micahl Thank you for all your insights.

Best regards.

**Comment by bogdan-patraucean** [comment]
This is still needed.

**Comment by yaira2** [comment]
This would be useful

**Comment by 0x5bfa** [comment]
I need it!!
